main/main.cpp

```diff
@@ -1855,9 +1855,23 @@ bool Main::start() {
 		} else {
 			return false;
 		}
-
-	} else {
-		main_loop_type = GLOBAL_DEF("application/run/main_loop_type", "");
+	} else { // Not based on script path.
+		if (!editor && !ClassDB::class_exists(main_loop_type) && ScriptServer::is_global_class(main_loop_type)) {
```

Comment:
```suggestion
		if (!ClassDB::class_exists(main_loop_type) && ScriptServer::is_global_class(main_loop_type)) {
```

Technically it's possible to override the main loop of the editor itself upon restart, but if there are any errors (wrong base type, invalid script etc), then the editor will always crash with an alert (you'd have to modify `project.godot` to revert to previous behavior). 😅